### PR TITLE
docs(api): hide certain private methods

### PR DESCRIPTION
**Pull request by ecormany**
# Overview

Hides a number of methods and parameters that aren't intended for (and likely aren't useful to) API users.

# Changelog

- Adds `:exclude-members:` entries to the API reference page.
- Does not use `:meta private:` in source docstrings because this is [not yet supported by autosummary](https://github.com/sphinx-doc/sphinx/issues/8922).

# Review requests

[Sandbox link](http://sandbox.docs.opentrons.com/RTC-40-hide-private-methods/v2/new_protocol_api.html)
- Check that all the methods that should be hidden are hidden.

# Risk assessment

v low

### CODE REVIEW

File: api/docs/v2/new_protocol_api.rst [hunk 1]
```diff
@@ -30,20 +30,22 @@ Modules
...
 
 .. autoclass:: opentrons.protocol_api.MagneticModuleContext
    :members:
+   :exclude-members: broker, geometry
    :inherited-members:
 
 .. autoclass:: opentrons.protocol_api.ThermocyclerContext
    :members:
-   :exclude-members: total_step_count, current_cycle_index, total_cycle_count, hold_time, ramp_rate, current_step_index, flag_unsafe_move
+   :exclude-members: total_step_count, current_cycle_index, total_cycle_count, hold_time, ramp_rate, current_step_index, flag_unsafe_move, broker, geometry
```
Comment (mcous) [inline api/docs/v2/new_protocol_api.rst]: `flag_unsafe_move` was just removed entirely by #11632, so can probably be removed here while we're at it

```suggestion
   :exclude-members: total_step_count, current_cycle_index, total_cycle_count, hold_time, ramp_rate, current_step_index, broker, geometry
```

**Review by mcous — APPROVED** [review]
✨

File: api/docs/v2/new_protocol_api.rst [hunk 2]
```diff
@@ -11,7 +11,7 @@ Protocols and Instruments
 
 .. autoclass:: opentrons.protocol_api.ProtocolContext
    :members:
-   :exclude-members: location_cache, _hw_manager
+   :exclude-members: location_cache, _hw_manager, cleanup, clear_commands, commands
 
 .. autoclass:: opentrons.protocol_api.InstrumentContext
```
Comment (mcous) [inline api/docs/v2/new_protocol_api.rst:16]: Anything in `InstrumentContext` worth adding, I wonder? Maybe `delay`, which is an accidental holdover from PAPIv1 and doesn't do anything

Comment (ecormany) [reply api/docs/v2/new_protocol_api.rst:16]: Seems good. We raise a pretty self-explanatory error for anyone still(!) trying to use this method: 
```
raise ExceptionInProtocolError(e, tb, str(e), frame.lineno) from e
opentrons.protocols.execution.errors.ExceptionInProtocolError: NotImplementedError [line 20]: InstrumentContext.delay() is not supported in Python Protocol API v2. Use ProtocolContext.delay() instead
```

Comment (ecormany) [reply api/docs/v2/new_protocol_api.rst:16]: Contrast `ProtocolContext.pause()` and `resume()`, which still _appear_ to work.

**Pull request merged**